src/main/java/org/gridsuite/caseimport/server/CaseImportService.java

```diff
@@ -14,29 +14,51 @@
 
 import java.util.UUID;
 
+import static org.gridsuite.caseimport.server.CaseImportException.Type.UNKNOWN_CASE_SOURCE;
+
 /**
  * @author Abdelsalem HEDHILI <abdelsalem.hedhili at rte-france.com>
  */
 @Service
 class CaseImportService {
 
-    @Value("${target-directory-name:automatic cases}")
+    @Value("${case-import-server.target-directories.default:automatic cases}")
     private String directoryName;
 
+    private final CaseImportConfig caseImportConfig;
+
     private final DirectoryService directoryService;
 
     private final CaseService caseService;
 
     static final String CASE = "CASE";
 
-    public CaseImportService(CaseService caseService, DirectoryService directoryService) {
+    public CaseImportService(CaseService caseService, DirectoryService directoryService, CaseImportConfig caseImportConfig) {
         this.caseService = caseService;
         this.directoryService = directoryService;
+        this.caseImportConfig = caseImportConfig;
+    }
+
+    private String getTargetDirectory(String caseOrigin) {
+        if (caseOrigin == null || caseOrigin.isEmpty()) {
+            return directoryName;
```

Comment:
The directoryName has an easy way to configure a default value with the @Value annotation in case it is not defined.
For a full blown configuration, I do not know how to do it. Any suggestions ?
We have a custom configuration that loads a mapping from configuration file. How would you make it so that if the "default" key does not exist, the default import directory is set to some value ?